os/arch/arm/src/amebad/amebad_i2c.c

```diff
@@ -0,0 +1,1237 @@
+#endif
+#ifdef CONFIG_I2C_RESET
+static int amebad_i2c_reset(FAR struct i2c_master_s *dev);
+#endif
+
+static int amebad_i2c_process(FAR struct i2c_dev_s *dev, FAR struct i2c_msg_s *msgs, int count);
+static uint32_t amebad_i2c_setfrequency(FAR struct i2c_dev_s *dev, uint32_t frequency);
+static int amebad_i2c_setaddress(FAR struct i2c_dev_s *dev, int addr, int nbits);
+static int amebad_i2c_write(FAR struct i2c_dev_s *dev, const uint8_t *buffer, int buflen);
+static int amebad_i2c_read(FAR struct i2c_dev_s *dev, uint8_t *buffer, int buflen);
+#ifdef CONFIG_I2C_WRITEREAD
+static int amebad_i2c_writeread(FAR struct i2c_dev_s *dev, const uint8_t *wbuffer, int wbuflen, uint8_t *buffer, int buflen);
+#endif
+
+/************************************************************************************
+ * Private Data
+ ************************************************************************************/
+
+/* Trace events strings */
+
+#ifdef CONFIG_I2C_TRACE
+static const char *g_trace_names[] = {
+	"NONE           ",
+	"SENDADDR       ",
+	"SENDBYTE       ",
+	"RCVBYTE        ",
+	"NOSTART        ",
+	"START/RESTART  ",
+	"STOP           ",
+	"ERROR          "
+};
+#endif
+
+/* I2C interface */
+
+static const struct i2c_ops_s amebad_i2c_ops = {
+	.setfrequency = amebad_i2c_setfrequency,
+	.setaddress = amebad_i2c_setaddress,
+	.write = amebad_i2c_write,
+	.read = amebad_i2c_read,
+#ifdef CONFIG_I2C_WRITEREAD
+	.writeread = amebad_i2c_writeread,
+#endif
+#ifdef CONFIG_I2C_TRANSFER
+	.transfer = amebad_i2c_transfer,
+#endif
+#ifdef CONFIG_I2C_SLAVE
+	.setownaddress = amebad_i2c_setaddress, //amebad_i2c_setownaddress,
+	//.registercallback = amebad_i2c_registercallback
+#endif
+};
+
+/* I2C device structures */
+
+static const struct amebad_i2c_config_s amebad_i2c1_config = {
+	//.base = AMEBAD_I2C1_BASE,
+	//.busy_idle = CONFIG_I2C1_BUSYIDLE,
+	//.filtscl = CONFIG_I2C1_FILTSCL,
+	//.filtsda = CONFIG_I2C1_FILTSDA,
+	.scl_pin = PA_25,
+	.sda_pin = PA_26,
+#ifndef CONFIG_I2C_SLAVE
+	.mode = AMEBAD_I2C_MASTER,
+#else
+	.mode = AMEBAD_I2C_SLAVE,
+#endif
+#ifndef CONFIG_I2C_POLLED
+	.irq = AMEBAD_IRQ_I2C0,
+#endif
+};
+
+static struct amebad_i2c_priv_s amebad_i2c1_priv = {
+	.ops = &amebad_i2c_ops,
+	.config = &amebad_i2c1_config,
+	.refs = 0,
+	.intstate = INTSTATE_IDLE,
+	.msgc = 0,
+	.msgv = NULL,
+	.ptr = NULL,
+	.dcnt = 0,
+	.flags = 0,
+	.status = 0
+};
+
+/************************************************************************************
+ * Private Functions
+ ************************************************************************************/
+
+/************************************************************************************
+ * Name: amebad_i2c_getreg
+ *
+ * Description:
+ *   Get a 32-bit register value by offset
+ *
+ ************************************************************************************/
+
+static inline uint32_t amebad_i2c_getreg(FAR struct amebad_i2c_priv_s *priv, uint16_t offset)
```

Comment:
remove this also. i think we are not using it and remove all those comments and API which are not helpful.